src/synthetic_knn/synthetic_plots.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import numpy as np
+import pandas as pd
+from numpy.typing import NDArray
+from sklearn.neighbors import NearestNeighbors
+
+from synthetic_knn.point_networks import PointNetwork
+
+
+class SyntheticPlots:
+    def __init__(
+        self, reference_coordinates: NDArray, network: PointNetwork, k: int = 10
+    ):
+        self.reference_coordinates = reference_coordinates
+        self.network = network
+        self.k = k
+        self.neighbors_ = self._generate_neighbors()
+
+    @staticmethod
+    def _create_column_names(n: int, prefix: str) -> list[str]:
+        return [f"{prefix}{i+1}" for i in range(n)]
+
+    def _create_dataframe(self, arr: NDArray, prefix: str = "NN") -> pd.DataFrame:
+        df = pd.DataFrame(
+            arr,
+            columns=self._create_column_names(arr.shape[1], prefix),
+            index=np.arange(1, arr.shape[0] + 1),
+        )
+        df.index.name = "SYNTHETIC_PLOT_ID"
+        return df
+
+    def synthetic_coordinates(self, as_frame: bool = False) -> NDArray:
+        coordinates = self.network.network_coordinates(self.reference_coordinates)
+        if as_frame:
+            return self._create_dataframe(coordinates, prefix="CCA")
```

Comment:
I missed it on my first look, but will this always be used with CCA space? I assume it could theoretically be used with other dimensionality reduction or even untransformed coordinates? 

Sorry if this throws a wrench into the works, but if you were to use this with untransformed coordinates, presumably you would want the column names to match the original feature names?

Maybe that's a rare enough use case that it's not worth worrying about, though.

Reply:
No, this is a great point and it did give me pause at some point and then I forgot about it.  To your previous point about moving `ipykernel` out of the package dependencies and into the hatch environment dependencies, I think this would (at present) hold with `sknnr` as well as there is no need to create the plot network from the plots that come out of `sknnr` (our tests don't currently rely on this).  The *notebooks* will still want to lean on `sknnr` for examples as that will be the most common use case, but I'm making the dependency change now.

I'm making `prefix` an optional keyword for `synthetic_coordinates` and defaulting it to the more generic `AXIS`.  If we were using `sknnr`, we could rely on getting back the axis names (for `CCA` and `CCorA`), but that seems like more trouble than it's worth at this point?  What do you think?

Reply:
> To your previous point about moving ipykernel out of the package dependencies and into the hatch environment dependencies, I think this would (at present) hold with sknnr as well as there is no need to create the plot network from the plots that come out of sknnr (our tests don't currently rely on this).

Good point, hadn't thought about that!

> I'm making prefix an optional keyword for synthetic_coordinates and defaulting it to the more generic AXIS.

:+1: 

>  If we were using sknnr, we could rely on getting back the axis names (for CCA and CCorA), but that seems like more trouble than it's worth at this point? What do you think?

Would you get the axis names by passing in the transformer to `SyntheticPlots`, or a dataframe output from the transformer, or something else? I had thought that `SyntheticPlots` could accept a dataframe for `reference_coordinates` and pull feature names from that, which would have the upside of preserving names if you wanted to create synthetic plots in an untransformed space. But I fully agree that this is more trouble than it's worth right now!

Reply:
> Would you get the axis names by passing in the transformer to `SyntheticPlots`, or a dataframe output from the transformer, or something else? I had thought that `SyntheticPlots` could accept a dataframe for `reference_coordinates` and pull feature names from that, which would have the upside of preserving names if you wanted to create synthetic plots in an untransformed space. But I fully agree that this is more trouble than it's worth right now!

Frankly, my comment was a bit too flippant and you point out some of the subtleties involved.  I agree with you that if `reference_coordinates` is a dataframe that it would make sense to pull feature names from it, since they would also be applicable to the `SyntheticPlots`.  As you point out, we don't want to rely explicitly on passing a transformer because that is not generic enough.

For now, I'm going to punt, but I will think about adding this as an issue.  Somewhat low priority for now.